src/org/omegat/filters2/moodlephp/MoodlePHPFilter.java

```diff
@@ -61,7 +61,7 @@
 
     public static final String OPTION_REMOVE_STRINGS_UNTRANSLATED = "unremoveStringsUntranslated";
 
-    protected static final Pattern RE_ENTITY = Pattern.compile("\\$string\\['(.+)'] (=) '(.+)';",
+    protected static final Pattern RE_ENTITY = Pattern.compile("\\$string\\['(.+)'\\] (=) '(.+)(';)$",
```

Comment:
Is this regex really sufficient? It looks like a [Moodle strings file](https://docs.moodle.org/dev/String_API#Adding_language_file_to_plugin) can be arbitrary PHP, so it could have whitespace or comments on the end. There's also some syntax involving HTML-like tags.

If you want to improve this, there is a common regex pattern for [matching only unescaped characters](https://stackoverflow.com/a/11819111/448068). This would let you match only the "real" end of the string literal and ignore anything afterwards.

Reply:
I do not develop very well. Sometimes I am able to adapt code to meet my needs (I rarely get it).

I don't know what php files are like. I needed Omegat to work natively with the Moodle php files. I made a filter that worked with Moodle files. But a translator using OmegaT found a failure in the filter.

Both pattern, the new and the previous, they work ok. The string is filtered before that the pattern executes. The error was in the while loop.

I will be very grateful if you can improve the filter.

Reply:
Sure, no problem. Thanks for your contribution.